tests/ci/lambda/src/bin/purge-stale-builds.rs

```diff
@@ -40,6 +42,8 @@ async fn handle(_event: LambdaEvent<Value>) -> Result<(), Error> {
 
     let project = std::env::var("CODEBUILD_PROJECT_NAME").unwrap();
 
+    let have_ec2_permissions: bool = project == "aws-lc-ci-ec2-test-framework";
```

Comment:
if you don't have permissions, should you just return here? what processing can you do without permissions?

Reply:
The operations related to `PrunedGitHubBuilds` can still be performed and the metrics should still be generated. (Most projects will have a 0-metric for `TerminatedEC2Instances`.) The problem was that the Lambda jobs for most of the projects (all except "aws-lc-ci-ec2-test-framework") were failing due to lack of EC2 permissions.